prow/cmd/pipeline/main.go

```diff
@@ -117,7 +117,9 @@ func main() {
 		logrus.WithError(err).Fatal("failed to load prow config")
 	}
 
-	configs, err := kube.LoadClusterConfigs(kube.NewConfig(kube.ConfigFile(o.kubeconfig)))
+	configs, err := o.kubernetes.LoadClusterConfigs(func() {
+		logrus.Fatal("Kubeconfig changed, exiting to trigger a restart")
```

Comment:
Why can't we load the kubeconfigs in memory instead of force exiting? Triggering a restart is only related if its running in a pod.

Reply:
This controller is not designed to pickup changes in kubeconfig. That would be next step of implementation.

Reply:
Fair enough.

/lgtm
/approve